src/snowcli/cli/common/flags.py

```diff
@@ -54,8 +63,89 @@
     "--schemaname",
     "--schema",
     help=" Schema in the database to use.",
+    callback=ConnectionDetails.update_callback("schema"),
+)
+
+
+RoleOption = typer.Option(
+    None,
+    "-r",
+    "--rolename",
+    "--role",
+    help="Role to be used.",
+    callback=ConnectionDetails.update_callback("role"),
 )
 
-RoleOption = typer.Option(None, "-r", "--rolename", "--role", help="Role to be used.")
+WarehouseOption = typer.Option(
+    None,
+    "-w",
+    "--warehouse",
+    help="Warehouse to use.",
+    callback=ConnectionDetails.update_callback("warehouse"),
+)
+
+
+def connection_params(func: Callable):
+    """
+    Decorator providing default flags for overriding connection parameters. Values are
+    updated in global SnowCLI state.
+
+    To use this decorator your command needs to accept **kwargs as last argument.
+    """
+    print("before", inspect.signature(func))
+
+    def wrapper(**kwargs):
+        return func(**kwargs)
 
-WarehouseOption = typer.Option(None, "-w", "--warehouse", help="Warehouse to use.")
+    sig = inspect.signature(func)
+    sig = sig.replace(
+        parameters=[
+            *tuple(sig.parameters.values())[:-1],  # Remove **kwargs from signature
+            # Connection flags
```

Comment:
I think this solution may work in all python versions supported by us (it's from [here](https://bugs.python.org/issue43355#msg388023)).
```suggestion
    type_hints = get_type_hints(func)
    sig = inspect.signature(func)
    existing_parameters = tuple(sig.parameters.values())[:-1]  # Remove **kwargs from signature
    existing_parameters_with_evaluated_types = [
        inspect.Parameter(
            name=p.name,
            kind=p.kind,
            annotation=type_hints[p.name],
            default=p.default,
        )
        for p in existing_parameters
    ]
    sig = sig.replace(
        parameters=[
            *existing_parameters_with_evaluated_types,
            # Connection flags    
```